src/qibolab/qubits.py

```diff
@@ -14,55 +11,35 @@ class Qubit(Model):
 
     Qubit objects are instantiated by :class:`qibolab.platforms.platform.Platform`
     but they are passed to instrument designs in order to play pulses.
-
-    Args:
-        name (int, str): Qubit number or name.
-        readout (:class:`qibolab.platforms.utils.Channel`): Channel used to
-            readout pulses to the qubit.
-        drive (:class:`qibolab.platforms.utils.Channel`): Channel used to
-            send drive pulses to the qubit.
-        flux (:class:`qibolab.platforms.utils.Channel`): Channel used to
-            send flux pulses to the qubit.
     """
 
     model_config = ConfigDict(frozen=False)
 
-    name: QubitId
-
-    probe: Optional[IqChannel] = None
-    acquisition: Optional[AcquireChannel] = None
-    drive: Optional[IqChannel] = None
-    drive12: Optional[IqChannel] = None
-    drive_cross: Optional[dict[QubitId, IqChannel]] = None
-    flux: Optional[DcChannel] = None
-
-    @property
-    def channels(self) -> Iterable[Channel]:
-        for ct in ChannelType:
-            channel = getattr(self, ct.value)
-            if channel is not None:
-                yield channel
+    drive: Optional[ChannelId] = None
+    """Ouput channel, to drive the qubit state."""
+    drive_qudits: dict[TransitionId, ChannelId] = Field(default_factory=dict)
+    """Output channels collection, to drive non-qubit transitions."""
```

Comment:
It looked to me more polished to have the general object.

Moreover, the other more subtle reason for doing this is to keep some "flexible space" for channels, to be abused.
It doesn't sound great, and if not needed it should not be used, but there might be occasions where it could be handy to associate extra channels to qubits in Qibocal. To avoid any possible breaking change in the short term, I left some room here, such that it could be used as a workaround (e.g., you can use the transition `0-1000`, `0-1001`, ... with a custom Qibocal encoding).
If any of these patterns will arise, we should then stabilize it with proper support. But possibly in 0.3. So, some extra wiggle room for such a huge change as 0.2 doesn't hurt.

Reply:
I assume there were no actual requests, just a comment. In case I misinterpreted, please reopen the conversation.

Reply:
Providing some flexible space sounds like a good idea. If that is the case, we could even make it more apparent that this is such a thing. For example, instead of the `tuple` (`TransitionId`) type, we could use just `str` for the keys and even have a less descriptive name, such as `additional`, instead of `drive_qudits`.

That being said, I am not sure if there are any "additional" channels that are not associated to some transition that could be relevant for some actual application. One example could be that we can do the 0->2 using one or two photons (with half the frequency) but I think that single photon 0->2 frequency cannot be reached by our electronics so it's probably not useful anyway.

Reply:
Possibly related to this discussion: what was implemented in #907 could be an example of an "additional" pulse that is not strictly associated to a drive to an excited state.

Reply:
Ah, yes, for that we should augment even the probe channels.

But I'd still keep as it is, for the time being. We're lacking person power for the full qudits project, and scheduling for 0.3 should not be that limiting.
The *extra wiggle room* here would still allow doing something like that, e.g. using the `0-1001` encoding for that is an option for the probe, since it's still an `IqChannel` (though you will lack a channel for the acquisition...).